iceoryx_posh/source/popo/ports/server_port_user.cpp

```diff
@@ -61,20 +71,72 @@ bool ServerPortUser::hasLostRequestsSinceLastCall() noexcept
 }
 
 cxx::expected<ResponseHeader*, AllocationError>
-ServerPortUser::allocateResponse(const uint32_t /*userPayloadSize*/) noexcept
+ServerPortUser::allocateResponse(const RequestHeader* const requestHeader,
+                                 const uint32_t userPayloadSize,
+                                 const uint32_t userPayloadAlignment) noexcept
 {
-    /// @todo
-    return cxx::error<AllocationError>(AllocationError::RUNNING_OUT_OF_CHUNKS);
+    cxx::Ensures(requestHeader != nullptr && "requestHeader must not be a nullptr");
+    if (requestHeader == nullptr)
+    {
+        // this branch will only be executed in tests where the error handler is suppressed
+        // and does not terminate with a fatal error
+        return cxx::error<AllocationError>(AllocationError::UNDEFINED_ERROR);
+    }
+
+    auto allocateResult = m_chunkSender.tryAllocate(
+        getUniqueID(), userPayloadSize, userPayloadAlignment, sizeof(ResponseHeader), alignof(ResponseHeader));
+
+    if (allocateResult.has_error())
+    {
+        return cxx::error<AllocationError>(allocateResult.get_error());
+    }
+
+    auto responseHeader =
+        new (allocateResult.value()->userHeader()) ResponseHeader(requestHeader->m_uniqueClientQueueId,
+                                                                  requestHeader->m_lastKnownClientQueueIndex,
+                                                                  requestHeader->getSequenceId());
+
+    return cxx::success<ResponseHeader*>(responseHeader);
 }
 
-void ServerPortUser::freeResponse(ResponseHeader* const /*responseHeader*/) noexcept
+void ServerPortUser::freeResponse(ResponseHeader* const responseHeader) noexcept
 {
-    /// @todo
+    cxx::Ensures(responseHeader != nullptr && "responseHeader must not be a nullptr");
+    if (responseHeader == nullptr)
+    {
+        return;
+    }
+    m_chunkSender.release(responseHeader->getChunkHeader());
 }
 
-void ServerPortUser::sendResponse(ResponseHeader* const /*responseHeader*/) noexcept
+void ServerPortUser::sendResponse(ResponseHeader* const responseHeader) noexcept
```

Comment:
Again, please use a reference to get rid of the `nullptr` check and additional error handling.